app.js

```diff
@@ -11,39 +13,98 @@ app.use(express.json());
 
 // Iteration 1 - Connect to MongoDB
 // DATABASE CONNECTION
+const MONGODB_URI = "mongodb://127.0.0.1:27017/express-mongoose-recipes-dev"
+
+mongoose.connect(MONGODB_URI)
+.then((x)=> console.log(`Connected to Mongo! Database name: "${x.connections[0].name}`))
+.catch((err) => console.log("Error connecting to Mongo", err))
 
 
 
 // ROUTES
-//  GET  / route
+//  GET  / route - This is just an example route
 app.get('/', (req, res) => {
-    res.send("<h1>Lab | Express Mongoose Recipes</h1>");
+    res.send("<h1>LAB | Express Mongoose Recipes</h1>");
 });
 
 
-//  Iteration 4 - Create recipe route
+//  Iteration 3 - Create a Recipe route
 //  POST  /recipes route
-
-
-//  Iteration 5 - Read all recipes
+app.post('/recipes', (req,res)=>{
+    console.log(req.body)
+    Recipe.create(req.body)
+    .then((createdRecipe)=>{
+        console.log(createdRecipe)
+        res.status(201).json(createdRecipe)
+    })
+    .catch((err)=>{
+        console.log(err)
+        res.status(500).json({errorMessage: "Error occured when creating a new recipe"})
+    })
+})
+
+
+//  Iteration 4 - Get All Recipes
 //  GET  /recipes route
-
-
-//  Iteration 6 - Read a single recipe
+app.get("/recipes", (req,res)=>{
+    Recipe.find()
+    .then((allRecipes)=>{
+        res.status(200).json(allRecipes)
+    })
+    .catch((err)=>{
+        console.log(err)
+        res.status(500).json({errorMessage: "Error occured when retrieving all recipe data"})
+    })
+})
+
+//  Iteration 5 - Get a Single Recipe
 //  GET  /recipes/:id route
-
-
-//  Iteration 7 - Update a single recipe
+app.get("/recipe/:id", (req,res)=>{
+    Recipe.findById(req.params.id)
```

Comment:
also applys for the id in the params 
```suggestion
const { id } = req.params
Recipe.findById(id)
```